src/lib/task_listener.js

```diff
@@ -331,7 +334,10 @@ export default class TaskListener extends EventEmitter {
     this.runningTasks.splice(taskIndex, 1);
     this.lastKnownCapacity += 1;
 
-    if (this.isIdle()) this.emit('idle', this);
+    if (this.isIdle()) {
```

Comment:
As you described we probably want to only call start idle work unless it's been idle for a certain amount of time since we call claimWork very frequently.